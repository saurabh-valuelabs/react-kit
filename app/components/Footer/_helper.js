/*
 *
 * Navigation helper
 *
 */

import React from 'react';

import { Nav, NavItem, NavLink } from 'reactstrap';
import { Link } from 'react-router-dom';

export function returnItem(item) {
  return item.childrenList.map(innerItem => (
    <NavItem key={`footer-link-${innerItem.name}`} className="col-auto">
      <NavLink
        tag={Link}
        to={innerItem.udi ? innerItem.udi : innerItem.url}
        className="px-0"
      >
        {innerItem.name}
      </NavLink>
    </NavItem>
  ));
}

export function createNav(item) {
  return item.map(innerItem => (
    <Nav key={`footer-title-${item.headingText}`} className="row">
      {returnItem(innerItem)}
    </Nav>
  ));
}
