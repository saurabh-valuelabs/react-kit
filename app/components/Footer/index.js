/**
 *
 * Footer
 *
 */

import React from 'react';
// import PropTypes from "prop-types";

import { Container, Row, Col } from 'reactstrap';

// import { createNav } from './_helper';

function Footer() {
  return (
    <Container fluid>
      <Row>
        <Col />
        <Col xs="auto">
          <span className="ml-auto px-0 nav-link">
            Powered by <a href="https://www.yoursite.com"> ValueLabs React</a>{' '}
          </span>{' '}
        </Col>{' '}
      </Row>{' '}
    </Container>
  );
}

Footer.propTypes = {};

export default Footer;
