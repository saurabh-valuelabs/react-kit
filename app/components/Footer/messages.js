/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Footer';

export default defineMessages({
  backToTop: {
    id: `${scope}.backToTop`,
    defaultMessage: 'Back To Top1',
  },
  backToTop2: {
    id: `${scope}.backToTop2`,
    defaultMessage: 'Back To Top1',
  },
});
