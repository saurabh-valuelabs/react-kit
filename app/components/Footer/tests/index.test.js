/**
 *
 * Tests for Footer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';

import { IntlProvider } from 'react-intl';
import footerNav from '../../../config/_footerNav';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Footer from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

jest.mock('react-router-dom', () => ({ Link: 'a' }));
describe('<Footer />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Footer nav={footerNav} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Footer nav={footerNav} />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
