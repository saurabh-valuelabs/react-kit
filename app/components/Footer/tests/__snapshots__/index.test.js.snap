// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render and match the snapshot 1`] = `
<footer>
  <hr />
  <div
    class="row"
  >
    <div
      class="col"
    />
    <div
      class="col-auto"
    >
      <button
        aria-label="Back To Top"
        class="btn btn-link"
        type="button"
      >
        <span>
          Back To Top
        </span>
      </button>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12 col-md-4 col-lg-2"
    >
      <h4>
        About Us
      </h4>
      <ul
        class="nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="px-0 nav-link"
            to="/privacy-policy"
          >
            Privacy Policy
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="px-0 nav-link"
            to="/list"
          >
            Term and Condition
          </a>
        </li>
      </ul>
    </div>
  </div>
</footer>
`;
